### PR TITLE
Improve project tooling

### PR DESCRIPTION
**Pull request by alorle**
* Update husky config
* Simplify and improve VSCode settings
* Improve Prettier config
* Improve Eslint config
* Add Stylelint

# <del>TODO:<del>
- [x] Remove programatic edition of `.eslintrc.json`
- [x] Fix https://github.com/leemonade/leemons/pull/471#issuecomment-2178169162

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -26,89 +33,24 @@
...
     }
   },
-  "plugins": [
-    "react",
-    "react-hooks",
-    "sonarjs",
-    "import",
-    "prettier"
-  ],
+  "plugins": ["react", "sonarjs", "import"],
```
Comment (MIGUELez11) [inline .eslintrc.json:44]: Why removing react-hooks linter rules? @alorle

Comment (alorle) [reply .eslintrc.json:44]: They are defined by the `extends` property. The [docs](https://www.npmjs.com/package/eslint-plugin-react-hooks) talk about `plugins` section only for avance usage.

Comment (alorle) [reply .eslintrc.json]: Seeing your comment, I have a doubt about the react plugin (not the react-hooks one) and the same thing happens, so I have also removed it from the list of plugins.

**Comment by alorle** [comment]
It would be necessary to see if the modifications made by the `leemons-react` package are still necessary. In my opinion not, since all the generated code is being ignored, therefore it would not be necessary to use the `import/no-unresolved` rule.
What do you think @MIGUELez11 ?

**Comment by alorle** [comment]
> > It would be necessary to see if the modifications made by the `leemons-react` package are still necessary. In my opinion not, since all the generated code is being ignored, therefore it would not be necessary to use the `import/no-unresolved` rule. What do you think @MIGUELez11 ?
> 
> @alorle I'll have a look, the `eslint` generated maybe not, but the `jsconfig` I guess it should for the new plugins to have the import aliases defined

Yeah, of course. The `jsconfig.json` is another thing...

**Comment by alorle** [comment]
@MIGUELez11 , when launching the frontend in dev mode there is an error:
```shell
✖ Compiled with errors.

Leemons is ready to be enjoyed!
  Local:            http://localhost:3000/
  On Your Network:  http://192.168.1.189:3000/

Note that the development build is not optimized.
To create a production build, use yarn front:build.

Failed to compile.

Module not found: Error: Can't resolve 'babel-loader' in '/Users/alvaroordunaleon/gh/leemons/apps/dev'
```

**Review by johan-fx — APPROVED** [review]

**Pull request merged**